src/Component/AutoMapper/Extractor/ReadAccessor.php

```diff
@@ -49,7 +49,7 @@ public function __construct(int $type, string $name, string $sourceClass = null,
      *
      * @throws CompileException
      */
-    public function getExpression(Expr\Variable $input): Expr
+    public function getExpression(Expr\Variable $input, string $fieldName): Expr
```

Comment:
not found of this new param, but for methods, `$this->name` is the method's name, but we must access the original property's name...

another way of doing this would be to add a `$originalFieldName` as a property of the class

Reply:
For better understanding of this,
this code is where we generate this kind of code :
```php
        if (\Jane\Component\AutoMapper\MapperContext::isAllowedAttribute($context, 'foo', $fieldvalue = $this->extractCallbacks['foo']($value))) {
            $result['foo'] = $fieldvalue;
        }
```

and the var `$fieldName` here is the one used in the array access `$this->extractCallbacks[**'foo'**]`, if we use `$this->name`, wed get `$this->extractCallbacks['getFoo']` which does not fit with the "extract callback" array keys

Reply:
Can't we split that in 2 fields:
- `name`, will contain the property name ;
- `accessor`, will contain the accessor of that property.

So you will have both information ?

Reply:
ok, I think I prefer this!